app/models/concerns/events/courses/state.rb

```diff
@@ -131,6 +132,14 @@ def send_application_closed_email
     Event::ApplicationClosedMailer.notice(self).deliver_later if groups.first.course_admin_email.present?
   end
 
+  def send_canceled_email
+    return if canceled_reason.nil?
+
+    leaders.each do |leader|
+      Event::CanceledMailer.send(canceled_reason, self, leader).deliver_later
```

Comment:
The Event Mailer and Multilinugal Mailer might be a bit too cleverly implemented: this will run the methods minimum_participants, no_leader or weather. However, I think it will only send mails to the leaders and not the participants. please also send the mails to the participants.